aea/configurations/base.py

```diff
@@ -1792,20 +1792,23 @@ def _create_or_update_from_json(
 
 
 def _compute_fingerprint(
-    package_directory: Path, ignore_patterns: Optional[Collection[str]] = None
+    package_directory: Path,
+    ignore_patterns: Optional[Collection[str]] = None,
+    is_recursive: bool = True,
+    ignore_directories: Optional[Collection[str]] = None,
 ) -> Dict[str, str]:
     ignore_patterns = ignore_patterns if ignore_patterns is not None else []
+    ignore_directories = ignore_directories if ignore_directories is not None else []
     ignore_patterns = set(ignore_patterns).union(DEFAULT_FINGERPRINT_IGNORE_PATTERNS)
     hasher = IPFSHashOnly()
     fingerprints = {}  # type: Dict[str, str]
     # find all valid files of the package
     all_files = [
         x
-        for x in package_directory.glob("**/*")
+        for x in package_directory.glob("**/*" if is_recursive else "*")
         if x.is_file()
-        and (
-            x.match("*.py") or not any(x.match(pattern) for pattern in ignore_patterns)
```

Comment:
Not sure why this was like that. ignore patterns were never evaluated for python files...